build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/CheckoutExternalGitDependency.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.IO;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.BuildTools.PrepTasks
+{
+	public class CheckoutExternalGitDependency : Git
+	{
+		[Required]
+		public ITaskItem ExternalGitDependency { get; set; }
+
+		public bool ShouldCleanDependency { get; set; } = true;
```

Comment:
Should this default to `true`?  Should we run `git clean -xdf` *ever*, without manual intervention?

Reply:
My thinking was that we could get into trouble if existing commercial components were already built in this tree. If so, it would in theory be possible for a commercial build to fail to rebuild against changes from a new commit, or against a set of xamarin-android dependencies that had also changed. This _should_ never be the case in CI though, so I am happy to flip this default value to false.

Reply:
I agree, this *should* never happen in CI.  This *could* happen on personal machines, which is why I'm wary of the entire idea: indiscriminately nuking build trees which may have local changes is Bad™.

I'd be more comfortable if this functionality didn't exist.  `git submodule foreach --recursive git clean -xdff` is already A Thing™.

Reply:
I've deleted the clean functionality outright, though the subsequent `checkout --force` also is likely to throw away local changes from an external reference... should I revisit this as well?

Reply:
Related historical conversation :-) : 
https://github.com/xamarin/xamarin-android/pull/1710#discussion_r189686597

During the PR iteration, the repo cloning that `build-tools/timing/timing.targets` performs used a `git reset --hard`, which is kinda/sorta like `git checkout --force`.  The final decision was to *not* use `git reset` at all.

Is this the same situation as before?  No.  Does the logic apply though?  Maybe?

The major point of PR #1710 is that "submodules" shouldn't be indiscriminately nuked without some explicit user behavior.  `make prepare`/`make prepare-commercial` constitutes explicit user behavior, so if we *only* do `git checkout --force`/whatever from those targets, it should be fine.

It would *not* be fine if `make all`/`make commercial` did that.

Reply:
Sounds good, marking as resolved as this code path should only execute during prep.